src/document/docUtil.ts

```diff
@@ -0,0 +1,53 @@
+import { SelectionItem } from '../store';
+
+/**
+ * Finds the closest ancestor DOM element with the specified class.
+ */
+export function findClosestElWithClass(element: Element | null, className: string): Element | null {
+    for (let el = element; el && el !== ((document as unknown) as Element); el = el.parentNode as Element | null) {
+        if (el.classList && el.classList.contains(className)) {
+            return el;
+        }
+    }
+
+    return null;
+}
+
+/**
+ * Returns the page element and page number that the element is on.
+ * If not found return null/-1
+ */
+export function getPageInfo(element: Element | null): { pageEl: Element | null; page: number } {
+    const pageEl = findClosestElWithClass(element, 'page') || null;
+    const page = parseInt((pageEl && pageEl.getAttribute('data-page-number')) || '-1', 10);
```

Comment:
A default value of `-1` seems confusing. What is the desired behavior? Also, can we use null coalescing operators to simplify this?

Reply:
The desired behavior is `If not found return null/-1`. I think `??` is the same as `||` here.

Reply:
It seems like we're using `-1` as a substitute for `null`. Should we use the latter, instead?